ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -103,35 +129,32 @@ def started(self, event: monitoring.CommandStartedEvent):
             if span is not None:
                 span.set_status(Status(StatusCanonicalCode.INTERNAL, str(ex)))
                 span.end()
-                self._remove_span(event)
+                self._pop_span(event)
 
     def succeeded(self, event: monitoring.CommandSucceededEvent):
         """ Method to handle a pymongo CommandSucceededEvent """
-        span = self._get_span(event)
-        if span is not None:
-            span.set_attribute(
-                "db.mongo.duration_micros", event.duration_micros
-            )
-            span.set_status(Status(StatusCanonicalCode.OK, event.reply))
-            span.end()
-            self._remove_span(event)
+        if not self.enable:
+            return
+        span = self._pop_span(event)
+        if span is None:
+            return
+        span.set_attribute("db.mongo.duration_micros", event.duration_micros)
+        span.set_status(Status(StatusCanonicalCode.OK, event.reply))
+        span.end()
 
     def failed(self, event: monitoring.CommandFailedEvent):
         """ Method to handle a pymongo CommandFailedEvent """
-        span = self._get_span(event)
-        if span is not None:
-            span.set_attribute(
-                "db.mongo.duration_micros", event.duration_micros
-            )
-            span.set_status(Status(StatusCanonicalCode.UNKNOWN, event.failure))
-            span.end()
-            self._remove_span(event)
-
-    def _get_span(self, event):
-        return self._span_dict.get(_get_span_dict_key(event))
-
-    def _remove_span(self, event):
-        self._span_dict.pop(_get_span_dict_key(event))
+        if not self.enable:
```

Comment:
I notice this snippet is almost identical to the above code. Don't know if it's worth refactoring into a utility method though.

Reply:
Do you mean the following snippet?
```
        if not self.is_enabled:
            return
```
?

Reply:
nvm. I maybe be going crazy, but I thought a diff verifying significantly more standard attributes like db type.